### PR TITLE
Update initial development

### PR DESCRIPTION
**Pull request by ricknjacky**
part of fix for #1 

git is now installed.

### CODE REVIEW

**Comment by ricknjacky** [comment]
@meganindya Please review this I have modified the Dockerfile for installation of Git and Emscripten 
please share feedback and suggestions. Thanks

**Review by meganindya — COMMENTED** [review]
Can you try compiling some dummy `.c`/`.cpp` file to `.wasm`?

You can try this to run `.wasm` here (https://webassembly.studio/) probably.

**Comment by ricknjacky** [comment]
> 
> 
> Can you try compiling some dummy `.c`/`.cpp` file to `.wasm`?
> 
> You can try this to run `.wasm` here (https://webassembly.studio/) probably.


Done.

**The output**
image

**Comment by ricknjacky** [comment]
The C file for the above output was:-

image

File: Dockerfile [hunk 1]
```diff
@@ -14,6 +14,15 @@ RUN npm install -g npm
 # install typescript compiler
 RUN npm install -g typescript
 
+# install git
+RUN apt-get -y install git
```
Comment (meganindya) [inline Dockerfile:18]:
```zsh
 => ERROR [ 5/11] RUN apt-get -y install git
------
 > [ 5/11] RUN apt-get -y install git:
#9 0.409 Reading package lists...
#9 0.419 Building dependency tree...
#9 0.420 Reading state information...
#9 0.422 E: Unable to locate package git
------
executor failed running [/bin/sh -c apt-get -y install git]: exit code: 100
```

Fixed it using
```Dockerfile
RUN apt update --no-install-recommends -yq \
    && apt-get install git -yq
```

File: Dockerfile [hunk 2]
```diff
@@ -14,6 +14,15 @@ RUN npm install -g npm
...
+
+# Install emsdk -- emscripten.
+RUN /tmp/emsdk_portable/emsdk update
```
Comment (meganindya) [inline Dockerfile:21]:
```zsh
 => ERROR [ 6/11] RUN /tmp/emsdk_portable/emsdk update
------
 > [ 6/11] RUN /tmp/emsdk_portable/emsdk update:
#9 0.629 /bin/sh: 1: /tmp/emsdk_portable/emsdk: not found
------
executor failed running [/bin/sh -c /tmp/emsdk_portable/emsdk update]: exit code: 127
```

**Review by meganindya — CHANGES_REQUESTED** [review]
I think this (https://emscripten.org/docs/getting_started/downloads.html) guide is the most reliable

**Comment by ricknjacky** [comment]
> 
> 
> I think this (https://emscripten.org/docs/getting_started/downloads.html) guide is the most reliable

thanks for the feedback, I'll look into this and investigate this `executor failed running [/bin/sh -c /tmp/emsdk_portable/emsdk update]: exit code: 127` and try to troubleshoot it.

**Comment by meganindya** [comment]
What machine are you on? I tried building this on `Docker version 20.10.2, build 2291f61` on `Windows 10 Home, build 19041`.

**Comment by ricknjacky** [comment]
> What machine are you on? I tried building this on `Docker version 20.10.2, build 2291f61` on `Windows 10 Home, build 19041`.

My local setup has the exact same credentials as yours. The issue of installing `emscripten` was misconstrued by me, I have now rectified the error, please review and I await the feedback on the same

File: Dockerfile [hunk 3]
```diff
@@ -14,6 +14,14 @@ RUN npm install -g npm
 # install typescript compiler
 RUN npm install -g typescript
 
+# install git
+RUN apt update --no-install-recommends -yq \
+    && apt-get install git -yq
+#install Emscripten
+FROM emscripten/emsdk:1.38.25
+# Install required tools that are useful for this project i.e. music-blocks
+ RUN apt update && apt install -y music-blocks
```
Comment (meganindya) [inline Dockerfile:23]: Could you explain this line?

Comment (ricknjacky) [reply Dockerfile:23]: image


It's a snippet from the official docker hub [page ](https://hub.docker.com/r/emscripten/emsdk) for emscripten.

Comment (meganindya) [reply Dockerfile:23]: There is no `music-blocks` in the `APT` repository

Comment (meganindya) [reply Dockerfile:23]: However, `FROM emscripten/emsdk:1.38.25` should fetch the `emscripten` layers. Pls check building the image and verify the `emcc` compilation of a `.c`/`.cpp` file to `.wasm`

**Pull request closed without merging**